app/src/main/java/mozilla/lockbox/presenter/EditItemPresenter.kt

```diff
@@ -81,15 +81,12 @@ class EditItemPresenter(
 
         view.deleteClicks
             .subscribe {
-                if (credentials != null) {
-                    dispatcher.dispatch(DataStoreAction.Delete(credentials!!))
-                    dispatcher.dispatch(RouteAction.ItemList)
-                } else {
-                    pushError(
+                credentials?.let { c ->
+                    dispatcher.dispatch(DialogAction.DeleteConfirmationDialog(c))
```

Comment:
Nit: For such a small use of `c` I think I would advocate for using the anonymous `it` name.

Reply:
Will fix this in #603 :)